### PR TITLE
lua: handle require'package.mod' during startup

### PR DESCRIPTION
**Pull request by bfredl**
ASAN gives a "stack redzone" error on a heap buffer, so I guess you can say things are getting pretty serious.

### CODE REVIEW

**Comment by bfredl** [comment]
This is unbreaking `require'myplugin'` when `myplugin` is a module. I need to test autoload also.

File: src/nvim/option.c [hunk 1]
```diff
@@ -3354,10 +3354,6 @@ did_set_string_option(
     if (!parse_winhl_opt(curwin)) {
       errmsg = e_invarg;
     }
-  } else if (varp == &p_rtp) {  // 'runtimepath'
-    if (!nlua_update_package_path()) {
-      errmsg = (char_u *)N_("E970: Failed to initialize lua interpreter");
-    }
```
Comment (tjdevries) [inline src/nvim/option.c:3360]: You're still going to make sure that when we update rtp, the Lua paths get updated immediately? (and in the same Lua evaluation?)

Comment (bfredl) [reply src/nvim/option.c:3360]: they are updated immediately per definition as the separate representation has been erased.

**Comment by bfredl** [comment]
Seems to work also for autoload but would be best to sync that part with upstream: https://github.com/vim/vim/pull/7193

**Comment by justinmk** [comment]
Is `:help lua-package-path` and `:help lua-require-example` still valid after this?

**Pull request merged**